gateway/src/main/java/com/mx/path/gateway/context/GatewayRequestContext.java

```diff
@@ -32,11 +39,25 @@ public final class GatewayRequestContext extends RequestContext {
    *
    * @return GatewayRequestContext
    */
+  @SuppressWarnings("PMD.EmptyCatchBlock")
   public static GatewayRequestContext current() {
     RequestContext requestContext = RequestContext.current();
     if (requestContext == null) {
       return null;
     }
+    if (requestContext.getOriginatingIP() == null) {
+      try {
+        URL url = new URL("https://api.ipify.org"); // Public IP service URL
+        HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+        urlConnection.setRequestMethod("GET");
+
+        BufferedReader in = new BufferedReader(new InputStreamReader(urlConnection.getInputStream(), StandardCharsets.UTF_8));
+        requestContext.setOriginatingIP(in.readLine());
+        in.close();
+      } catch (IOException e) {
+        //Do nothing as we don't want to obstruct the flow
+      }
+    }
```

Comment:
I wonder if it would be better to put this in the getOringiatingIP getter in RequestContext. ?

Reply:
Yea I moved it to the getter. Thanks